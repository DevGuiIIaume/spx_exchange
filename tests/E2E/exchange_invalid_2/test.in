[T0] CANCEL 0;
[T0] BUY 0 Carrot 10 500;
[T0] CANCEL 0;
[T0] BUY 0 Router 50 100;
[T1] CANCEL 0;
[T0] CANCEL 0;
[T1] SELL 0 Router 50 100;
[T0] CANCEL 0;
[T0] BUY 1 Router 100 100;
[T1] CANCEL 1;
[T0] CANCEL 1;
[T0] DISCONNECT;
[T1] DISCONNECT;
