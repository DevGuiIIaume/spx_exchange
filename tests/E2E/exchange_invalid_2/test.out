[SPX] Starting
[SPX] Trading 2 products: GPU Router
[SPX] Created FIFO /tmp/spx_exchange_0
[SPX] Created FIFO /tmp/spx_trader_0
[SPX] Connected to /tmp/spx_exchange_0
[SPX] Connected to /tmp/spx_trader_0
[SPX] Created FIFO /tmp/spx_exchange_1
[SPX] Created FIFO /tmp/spx_trader_1
[SPX] Connected to /tmp/spx_exchange_1
[SPX] Connected to /tmp/spx_trader_1
[SPX] [T0] Parsing command: <CANCEL 0>
[SPX] [T0] Parsing command: <BUY 0 Carrot 10 500>
[SPX] [T0] Parsing command: <CANCEL 0>
[SPX] [T0] Parsing command: <BUY 0 Router 50 100>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 1; Sell levels: 0
[SPX]		BUY 50 @ $100 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <CANCEL 0>
[SPX] [T0] Parsing command: <CANCEL 0>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 0; Sell levels: 0
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <SELL 0 Router 50 100>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 0; Sell levels: 1
[SPX]		SELL 50 @ $100 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX]	Trader 1: GPU 0 ($0), Router 0 ($0)
[SPX] [T0] Parsing command: <CANCEL 0>
[SPX] [T0] Parsing command: <BUY 1 Router 100 100>
[SPX] Match: Order 0 [T1], New Order 1 [T0], value: $5000, fee: $50.
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 1; Sell levels: 0
[SPX]		BUY 50 @ $100 (1 order)
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 50 ($-5050)
[SPX]	Trader 1: GPU 0 ($0), Router -50 ($5000)
[SPX] [T1] Parsing command: <CANCEL 1>
[SPX] [T0] Parsing command: <CANCEL 1>
[SPX]	--ORDERBOOK--
[SPX]	Product: GPU; Buy levels: 0; Sell levels: 0
[SPX]	Product: Router; Buy levels: 0; Sell levels: 0
[SPX]	--POSITIONS--
[SPX]	Trader 0: GPU 0 ($0), Router 50 ($-5050)
[SPX]	Trader 1: GPU 0 ($0), Router -50 ($5000)
[SPX] Trader 0 disconnected
[SPX] Trader 1 disconnected
[SPX] Trading completed
[SPX] Exchange fees collected: $50
